ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -140,7 +145,12 @@ public int validate(
     if (customValidator != null) {
       customValidator.validate(configured, executionContext, serviceContext);
     } else if (KsqlEngine.isExecutableStatement(configured.getStatement())) {
-      executionContext.execute(injector.inject(configured));
+      final ConfiguredStatement<?> statementInjected = injector.inject(configured);
+
+      topicAccessValidator.apply(serviceContext, executionContext.getMetaStore())
```

Comment:
For CSAS, the sink topic is created when the line `injector.inject(configured)` is called (just above this line). It is created using the user credentials so it will fail if the user cannot create topics. I leverage the injector to validate some user permissions by letting it create topics, read some topics and/or write too (like insert into values).

I did run several tests in my local environment to make sure I have denied, unauthorized or unexistent topics, and they all work.

Reply:
That all makes sense, but my question still stands I think. `TopicAccessValidator` looks for the sink source in the meta-store, but it won't actually be in the meta-store, because its added to the meta-store by the engine on the next line. Which I think actually works fine, since internally we add the topic name to CREATE STREAM as a property before calling `topicAccessValidator`, and therefore `topicAccessValidator` never looks in the meta-store.